src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -204,35 +205,44 @@ impl Config {
 
         #[cfg(test)]
         // Check that the correct multiple is obtained.
+        // This inlined test is only done for valid 64-bit magnitudes
+        // and valid +/- 1 signs.
+        // Invalid values result in constraint failures which are
+        // tested at the circuit-level.
         {
             use group::Curve;
 
-            let base: super::OrchardFixedBases = base.clone().into();
-
-            let scalar =
-                scalar
-                    .magnitude
-                    .value()
-                    .zip(scalar.sign.value())
-                    .map(|(magnitude, sign)| {
-                        // Move magnitude from base field into scalar field (which always fits
-                        // for Pallas).
-                        let magnitude = pallas::Scalar::from_bytes(&magnitude.to_bytes()).unwrap();
-
-                        let sign = if sign == pallas::Base::one() {
-                            pallas::Scalar::one()
-                        } else if sign == -pallas::Base::one() {
-                            -pallas::Scalar::one()
-                        } else {
-                            panic!("Sign should be 1 or -1.")
-                        };
-                        magnitude * sign
-                    });
-            let real_mul = scalar.map(|scalar| base.generator() * scalar);
-            let result = result.point();
-
-            if let (Some(real_mul), Some(result)) = (real_mul, result) {
-                assert_eq!(real_mul.to_affine(), result);
+            if let (Some(magnitude), Some(sign)) = (scalar.magnitude.value(), scalar.sign.value()) {
+                let magnitude_is_valid =
+                    magnitude < pallas::Base::from_u64(0xFFFF_FFFF_FFFF_FFFFu64);
```

Comment:
This bound is slightly too small:
```suggestion
                    magnitude <= pallas::Base::from_u64(0xFFFF_FFFF_FFFF_FFFFu64);
```
(Non-critical because this is only test code, and if someone _did_ set a maximal magnitude this test would just be skipped.)